src/source/Ice/TurnConnection.c

```diff
@@ -418,14 +408,11 @@ STATUS turnConnectionHandleStunError(PTurnConnection pTurnConnection, PBYTE pBuf
                   pStunAttributeErrorCode->errorCode, pStunAttributeErrorCode->attribute.length, pStunAttributeErrorCode->errorPhrase);
 
             /* Find TurnPeer using transaction Id, then mark it as failed */
-            doubleListGetHeadNode(pTurnConnection->turnPeerList, &pCurNode);
-            while (pCurNode != NULL) {
-                pTurnPeer = (PTurnPeer) pCurNode->data;
-                pCurNode = pCurNode->pNext;
+            for (i = 0; i < pTurnConnection->turnPeerCount; ++i) {
+                pTurnPeer = &pTurnConnection->turnPeerList[i];
                 if (transactionIdStoreHasId(pTurnPeer->pTransactionIdStore, pBuffer + STUN_PACKET_TRANSACTION_ID_OFFSET)) {
                     pTurnPeer->connectionState = TURN_PEER_CONN_STATE_FAILED;
-                    /* break the loop */
-                    pCurNode = NULL;
+                    break;
```

Comment:
I strongly recommend not using condition flow modification keywords inside switch statements.